releaseinfo/registryclient/client.go

```diff
@@ -0,0 +1,126 @@
+package registryclient
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/docker/distribution/registry/client/transport"
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/hypershift/releaseinfo/registryclient/dockerarchive"
+	"github.com/openshift/hypershift/releaseinfo/registryclient/dockercredentials"
+	"github.com/openshift/hypershift/releaseinfo/registryclient/manifest"
+	"github.com/openshift/hypershift/releaseinfo/registryclient/reference"
+	"github.com/openshift/hypershift/releaseinfo/registryclient/registryclient"
+)
+
+// ExtractImageFiles extracts a list of files from a registry image given the image reference, pull secret and the
+// list of files to extract. It returns a map with file contents or an error.
+func ExtractImageFiles(imageRef string, pullSecretFile string, files ...string) (map[string][]byte, error) {
```

Comment:
Is this new code or copied? It should accept a `Context` instead of hard-coding `context.Background`... this is important for us because we need to be careful not to block our reconcile goroutines